src/test/java/fi/okm/jod/yksilo/controller/koski/IntegraatioKoskiControllerTest.java

```diff
@@ -208,4 +222,100 @@ void shouldReturnNoDataError_whenUserHaveNoDataInKoski() throws Exception {
 
     verify(koskiOAuth2Service).fetchDataFromResourceServer(oAuth2AuthorizedClient);
   }
+
+  @WithUserDetails("test")
+  @Test
+  void shouldReturnOsaamisetIdentified_whenValidRequestProvided() throws Exception {
+    var koulutusUUIDs =
+        List.of(
+            UUID.fromString("5edaca37-8ca1-4f18-918b-7aa73997c676"),
+            UUID.fromString("842a5528-06ac-455a-8d7e-7a401947b1f7"));
+    var yksilo = new Yksilo(UUID.nameUUIDFromBytes("test".getBytes()));
+
+    var koulutus1 = mock(Koulutus.class);
+    when(koulutus1.getYksilo()).thenReturn(yksilo);
+    when(koulutus1.getId()).thenReturn(koulutusUUIDs.get(0));
+    when(koulutus1.getNimi()).thenReturn(new LocalizedString(Map.of(Kieli.FI, "koulutus1")));
+    when(koulutus1.getAlkuPvm()).thenReturn(LocalDate.of(2023, 1, 1));
+    when(koulutus1.getLoppuPvm()).thenReturn(LocalDate.of(2023, 12, 31));
+    when(koulutus1.getOsaamisenTunnistusStatus()).thenReturn(OsaamisenTunnistusStatus.DONE);
+    var yksilonOsaaminen1 = mock(YksilonOsaaminen.class);
+    var osaaminen1 = mock(Osaaminen.class);
+    when(osaaminen1.getUri())
+        .thenReturn(
+            URI.create("http://data.europa.eu/esco/skill/008fa98b-dba6-4abf-909e-04299728e3eb"));
+    when(yksilonOsaaminen1.getOsaaminen()).thenReturn(osaaminen1);
+    when(koulutus1.getOsaamiset()).thenReturn(Set.of(yksilonOsaaminen1));
+    var koulutus2 = mock(Koulutus.class);
+    when(koulutus2.getYksilo()).thenReturn(yksilo);
+    when(koulutus2.getId()).thenReturn(koulutusUUIDs.get(1));
+    when(koulutus2.getNimi()).thenReturn(new LocalizedString(Map.of(Kieli.FI, "koulutus2")));
+    when(koulutus2.getAlkuPvm()).thenReturn(LocalDate.of(2024, 1, 1));
+    when(koulutus2.getOsaamisenTunnistusStatus()).thenReturn(OsaamisenTunnistusStatus.WAIT);
+    when(koulutusRepository.findByKokonaisuusYksiloIdAndIdIn(yksilo.getId(), koulutusUUIDs))
```

Comment:
Since this test tests the controller, it should mock (at most) the service which is a dependency of the controller, and not depend on that the service itself is implemented in a certain way.